website/content/specification/syntax/constraints.md

```diff
@@ -163,6 +163,17 @@ And the following document.
 
 The expect constraint would pass for each `sibling` in the `parent` named "p1", and would fail for each `sibling` in the `parent` named "p2".
 
+## `expect` constraints
+
+The `expect` constraint is a type of Metaschema constraint that restricts field or flag value(s) based on the evaluation of a test Metapath expression.
+
+The `@target` attribute of an `<expect>` constraint specifies the node(s) in a document instance whose value is restricted by the constraint.
+
+The `@test` attribute of an `<expect>` constraint specifies the logical condition to be evaluated against the document instance node(s) of the `@target`. This expression must evaluate to [a Metaschema boolean type values](/specification/datatypes#boolean) true or false.
+
+When the `@test` expression evaluates to true, the document instance node(s) are valid and conform to the constraint.
+
+When the `@test` expression evaluates to false, the document instance node(s) are not valid and not conform to the constraint. A constraint may have an optional [`@level`](#level) attribute and/or an optional child `<message>` element to indicate severity and documentation explaining how the target nodes are invalid.
```

Comment:
How should we handle validation errors? i.e. what happens when the test fails for a given value?

Also, the message can contain placeholder values using the `{metapath}` notation. We should document that.

Should this be swapped with [mustash ](https://mustache.github.io/mustache.5.html) style syntax? i.e. `{{metapath}}`?

Reply:
> How should we handle validation errors? i.e. what happens when the test fails for a given value?

Excellent you ask, I spent a lot of time on this simple edit because I was looking at the current [`metschema.xsd`](https://raw.githubusercontent.com/usnistgov/metaschema/develop/schema/xml/metaschema.xsd), Java, XSLT, OSCAL model definitions, and intuition. As it stands, I believe the default level if unset is `ERROR`, not `INFORMATIONAL`, so does this mean it should just error out without a custom message (by virtue of no `<message/>`)? Is that too reductive or am I sussing out the right level of detail here?

> Also, the message can contain placeholder values using the {metapath} notation. We should document that.
>
> Should this be swapped with [mustash ](https://mustache.github.io/mustache.5.html)style syntax? i.e. {{metapath}}?

Ironically, I was unaware of the `{{metapath}}` Mustache syntax was an option. I guess I need to talk to you further about that.

Reply:
I will separate that out and add in the Common Constraint Data section shortly in #496.

Reply:
@david-waltermire can you look at #497 which will go in the section above this and tell me if that is good? If so, does that still warrant some changes here?

(Re why we have message in expect too so I can adjust to point up to the one explanation of this useful feature and not repeatedly copy-paste it for applicable constraint types.)

Reply:
Also, I did not explain the rational for "keeping it as-is" based on previous conversations in Gitter: I assume if we want Metapath to continue closely matching XPath 3.x for understanding and lessen implementation burden, the current [Attribute Value Syntax is with single braces](http://dh.obdurodon.org/avt.xhtml), although the more I think about it is not ironically a XPath feature, but a XSLT 2.0/3 feature with XPath data evaluated from XPath/XSLT variables? Either way, happy with keeping it as-is and not adding Mustache into the mix, but happy to hear what community members think.

Reply:
Woops, scratch that. Sorry for the issue/PR pasta. I figured out I was wrong re the comment and linked analysis in https://github.com/usnistgov/metaschema/issues/496#issuecomment-1872347182. Added all the message docs specific to expect here, resolving for now.